api/src/main/java/jakarta/data/restrict/Restrict.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.restrict;
+
+import java.util.List;
+import java.util.Set;
+
+import jakarta.data.restrict.Restriction.Basic;
+import jakarta.data.restrict.Restriction.Operator;
+
+// TODO document
+// This is one of two places from which to obtain restrictions.
+// The other place is from static metamodel attributes.
+public enum Restrict {
+    ALL,
+    ANY;
+
+    @SafeVarargs
+    public static <T> Restriction<T> all(Restriction<T>... restrictions) {
+        return new CompositeRestriction<>(ALL, List.of(restrictions));
+    }
+
+    @SafeVarargs
+    public static <T> Restriction<T> any(Restriction<T>... restrictions) {
+        return new CompositeRestriction<>(ANY, List.of(restrictions));
+    }
```

Comment:
Still no way to do `not()`, is that correct?

Reply:
Because it requires further discussion, I chose to defer it and leave a TODO comment (right after this) for it.  A follow-on PR can add it after we decide how best to do it.

Reply:
Yeah no worries, was just checking I had not missed it.